### PR TITLE
Unhardcode Cursor definitions.

### PR DESCRIPTION
**Pull request by LipkeGu**
Unhardcode the Cursors and documentate them.

### CODE REVIEW

File: OpenRA.Mods.Common/Orders/DeployOrderTargeter.cs [hunk 1]
```diff
@@ -16,10 +16,13 @@ namespace OpenRA.Mods.Common.Orders
 {
 	public class DeployOrderTargeter : IOrderTargeter
 	{
+		[Desc("The Cursor that appears when units can be deployed.")]
+		public static readonly string DeployUnitCursor = "deploy";
```
Comment (pchote) [inline OpenRA.Mods.Common/Orders/DeployOrderTargeter.cs]: `DeployCursor` please.

**Comment by penev92** [comment]
I'm all for unhardcoding stuff.
But these aren't traits, so the values won't be read from YAML anyway. You might as well make them `const`, and also `private`.
`Desc` is used for trait documentation generation, so it won't help here. Might be better to switch those to regular comments to avoid confusion. Other than that I like the idea.

**Comment by pchote** [comment]
The names are self-descriptive, so the comments/descs aren't needed at all.

File: OpenRA.Game/Orders/UnitOrderGenerator.cs [hunk 2]
```diff
@@ -17,6 +17,12 @@ namespace OpenRA.Orders
 {
 	class UnitOrderGenerator : IOrderGenerator
 	{
+		[Desc("The Cursor that appears when Actors can be selected.")]
+		public static readonly string SelectActorCursor = "select";
```
Comment (pchote) [inline OpenRA.Game/Orders/UnitOrderGenerator.cs]: `SelectCursor` please.

**Comment by LipkeGu** [comment]
updated

**Comment by Phrohdoh** [comment]
This isn't actually unhardcoding anything. You'd need to setup default values in the constructors that you then override for this to actually make a difference to modders.

File: OpenRA.Game/Orders/UnitOrderGenerator.cs [hunk 3]
```diff
@@ -17,6 +17,9 @@ namespace OpenRA.Orders
 {
 	class UnitOrderGenerator : IOrderGenerator
 	{
+		private const string SelectCursor = "select";
```
Comment (pchote) [inline OpenRA.Game/Orders/UnitOrderGenerator.cs]: You don't need to (and shouldn't) explicitly write `private`.

Comment (Phrohdoh) [inline OpenRA.Game/Orders/UnitOrderGenerator.cs]: This also ought not be `const` if you want it to be configurable in the way I stated earlier in a top-level comment.

Comment (LipkeGu) [inline OpenRA.Game/Orders/UnitOrderGenerator.cs]: ok then im set them back to public

Comment (Phrohdoh) [inline OpenRA.Game/Orders/UnitOrderGenerator.cs]: My point is that you can pass these in through the ctor (using default values) via traits that create this.

That is how you can read the values in from YAML.

Comment (LipkeGu) [inline OpenRA.Game/Orders/UnitOrderGenerator.cs]: ah ok :)

**Comment by obrakmann** [comment]
Closing due to inactivity. Also, as @Phrohdoh mentioned, this doesn't unhardcode the cursor definitions in a sense that would be useful, ie. exposing them to yaml.

**Comment by Mailaender** [comment]
Yep, those should be moved to `TraitInfo` classes instead otherwise they aren't exposed to documentation. Then you could also add a `CursorReference` to them and add a `CheckCursors` lint check.

**Pull request closed without merging**